### PR TITLE
[CIRC-1347] Integrate ItemsInTransitReportService into ItemsInTransitResourse

### PR DESCRIPTION
**Pull request by OleksandrVidinieiev**

### CODE REVIEW

File: src/main/java/org/folio/circulation/domain/representations/ItemsInTransitReport.java [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+      .put("items", new JsonArray(reportEntries))
+      .put("totalRecords", reportEntries.size());
+  }
+
+  private Comparator<Item> sortByCheckinServicePointComparator() {
+    return comparing(item -> ofNullable(reportContext.getLoans().get(item.getItemId()))
+      .flatMap(loan -> ofNullable(loan.getCheckInServicePointId())
+        .flatMap(servicePointId -> ofNullable(reportContext.getServicePoints().get(servicePointId)))
+        .map(ServicePoint::getName))
+      .orElse(null), Comparator.nullsLast(String::compareTo));
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/circulation/domain/representations/ItemsInTransitReport.java:51]:
```
    return comparing(item -> ofNullable(reportContext.getLoans().get(item.getItemId()))
      .map(Loan::getCheckInServicePointId)
      .map(id -> reportContext.getServicePoints().get(id))
      .map(ServicePoint::getName)
      .orElse(null), Comparator.nullsLast(String::compareTo));
```

File: src/main/java/org/folio/circulation/domain/representations/ItemsInTransitReport.java [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+      return new JsonObject();
+    }
+
+    Holdings holdings = reportContext.getHoldingsRecords().get(item.getHoldingsRecordId());
+    if (holdings != null) {
+      Instance instance = reportContext.getInstances().get(holdings.getInstanceId());
+      if (instance != null) {
+        item = item.withInstance(instance);
+      }
+    }
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/circulation/domain/representations/ItemsInTransitReport.java:65]: Same can be achieved with Optional:
```
    item = ofNullable(item.getHoldingsRecordId())
      .map(id -> reportContext.getHoldingsRecords().get(id))
      .map(Holdings::getInstanceId)
      .map(id -> reportContext.getInstances().get(id))
      .map(item::withInstance)
      .orElse(item);
```

File: src/main/java/org/folio/circulation/infrastructure/storage/requests/RequestRepository.java [hunk 3]
```diff
@@ -52,6 +53,20 @@
   private final ServicePointRepository servicePointRepository;
   private final PatronGroupRepository patronGroupRepository;
 
+  /**
+   * Public constructor to avoid creating repositories twice
+   */
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/circulation/infrastructure/storage/requests/RequestRepository.java:58]: It's not clear from the comment where we risk instantiating repositories twice

File: src/main/java/org/folio/circulation/infrastructure/storage/requests/RequestRepository.java [hunk 4]
```diff
@@ -149,8 +164,11 @@ public static RequestRepository using(org.folio.circulation.support.Clients clie
   public CompletableFuture<Result<MultipleRecords<Request>>> findOpenRequestsByItemIds(
     Collection<String> itemIds) {
 
+    Result<CqlQuery> query = exactMatchAny("status", openStates())
+      .map(q -> q.sortBy(ascending("position")));
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/circulation/infrastructure/storage/requests/RequestRepository.java:168]: Do we really need to sort them? We save requests in a HashMap afterwards anyway

**Review by alexanderkurash — APPROVED** [review]

**Pull request merged**